active/IndexPath/IndexPath.md

```diff
@@ -0,0 +1,96 @@
+# Background
+
+To easily track selection in nested collections, we need an object that we can use to uniquely identify an item in nested collections.
+This can be done with the IndexPath.
+
+# Description
+
+The `IndexPath` class provides indexing of nested collections, for example in a TreeView.
+
+# Examples
+
+In the tree below, using the following IndexPath 
+```c#
+
+var indexPath = IndexPath.CreateFromIndices(new List<int>{ 1, 0, 3 });
+
+```
+ would select the item "Kitchen cabinet style", since it is in the second group of first nesting level, first group in the second level of nesting and is the fourth element in the last nesting level.
+
+![Sample tree selection](./sample-tree-selection.png)
+
+### Creating IndexPath objects
+New IndexPaths objects can be created using the static `IndexPath.CreateFrom` and `IndexPath.CreateFromIndices` methods:
+
+```csharp
+
+// Points to the third element in a flat collection
+var simplePath = IndexPath.CreateFrom(2);
+
+// This is the same IndexPath as "simplePath"
+var simplePathIndices = IndexPath.CreateFromIndices(new List<int>(){ 2 });
+
+// This will point to the third child of the second element in a nested collection
+var groupedPath = IndexPath.CreateFrom(1,2);
+
+// The following IndexPath is equivalent to "groupedPath"
+var groupedPathIndices = IndexPath.CreateFromIndices(new List<int>(){ 1, 2 });
+
+```
+
+### Comparing two IndexPaths
+Comparing two `IndexPath` objects can be done using the `CompareTo` function:
+
+```c#
+var baseIndexPath = IndexPath.CreateFromIndices(new List<int> { 1, 4, 2 });
+
+var shorterPath = IndexPath.CreateFrom(2, 0);
+
+// Since shorterPath takes second element of first nesting, it is "greater" than baseIndexPath
+Assert.AreEqual(1, shorterPath.CompareTo(baseIndexPath));
+
+
+var sameLengthPath = IndexPath.CreateFromIndices(new List<int> { 1, 0, 1 });
+
+// Since sameLength path enters before baseIndexPath on the second level, it is smaller than basePath
+Assert.AreEqual(-1, sameLengthPath.CompareTo(baseIndexPath));
+
+
+var baseIndexPathCopy = IndexPath.CreateFromIndices(new List<int> { 1, 4, 2 });
+// Those two are the same, and thus CompareTo returns 0
+Assert.AreEqual(0, baseIndexPathCopy.CompareTo(baseIndexPath));
+```
+
+# API Notes
+
+| Function | Description |
+|-|-|
+|Int32 GetSize()| Returns the size of the IndexPath. |
```

Comment:
Does size refer to the total amount of nodes you have to step through to get to the item referenced by the IndexPath calling this function, or does it refer to the # of levels?

Reply:
That refers to the number of levels.

Reply:
Got it, may want to clarify in the API Notes unless it's mentioned somewhere else.

Reply:
I've updated the API description of that and the GetAt function to be clearer.